tsinfer/formats.py

```diff
@@ -2412,16 +2455,18 @@ def __init__(
                 )
         self._sites_ancestral_allele = self._sites_ancestral_allele.astype(str)
         unknown_alleles = collections.Counter()
-        converted = np.zeros(self.num_sites, dtype=np.int8)
+        converted = np.full(self.num_sites, -1, dtype=np.int8)
         for i, allele in enumerate(self._sites_ancestral_allele):
-            allele_index = -1
-            try:
-                allele_index = np.where(allele == self.sites_alleles[i])[0][0]
-            except IndexError:
-                unknown_alleles[allele] += 1
-            converted[i] = allele_index
+            if not (allele in {"", "N", "n"}):  # All these must represent unknown
```

Comment:
We shouldn't treat empty string as legal input here as it has a defined meaning in VCZ

Reply:
Now done. See comment below about changing the name to "ancestral_state", to match what we used previously.